language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/InvocationNodeContextProvider.java

```diff
@@ -110,18 +111,40 @@ public void sort(BallerinaCompletionContext context, T node, List<LSCompletionIt
         }
 
         List<String> existingNamedArgs = CommonUtil.getDefinedArgumentNames(context, params.get(), argumentNodeList);
-        Predicate<ParameterSymbol> predicate = parameter -> parameter.paramKind() == ParameterKind.REQUIRED
-                || parameter.paramKind() == ParameterKind.DEFAULTABLE;
-        params.get().stream().filter(predicate).forEach(parameterSymbol -> {
-            Optional<String> paramName = parameterSymbol.getName();
-            TypeSymbol paramType = parameterSymbol.typeDescriptor();
-            String defaultValue = CommonUtil.getDefaultValueForType(paramType).orElse("");
-            if (paramName.isEmpty() || paramName.get().isEmpty() || existingNamedArgs.contains(paramName.get())) {
-                return;
+        for (ParameterSymbol parameterSymbol : params.get()) {
+            if (parameterSymbol.paramKind() == ParameterKind.REQUIRED ||
+                    parameterSymbol.paramKind() == ParameterKind.DEFAULTABLE) {
+                Optional<String> paramName = parameterSymbol.getName();
+                TypeSymbol paramType = parameterSymbol.typeDescriptor();
+                String defaultValue = CommonUtil.getDefaultValueForType(paramType).orElse("");
+                if (paramName.isEmpty() || paramName.get().isEmpty() || existingNamedArgs.contains(paramName.get())) {
+                    continue;
+                }
+                CompletionItem completionItem = NamedArgCompletionItemBuilder.build(paramName.get(), defaultValue);
+                completionItems.add(
+                        new NamedArgCompletionItem(context, completionItem, Either.forLeft(parameterSymbol)));
+            } else if (parameterSymbol.paramKind() == ParameterKind.INCLUDED_RECORD) {
+                TypeSymbol typeSymbol = CommonUtil.getRawType(parameterSymbol.typeDescriptor());
+                if (typeSymbol.typeKind() != TypeDescKind.RECORD) {
+                    // Impossible
+                    continue;
+                }
+                RecordTypeSymbol includedRecordType = (RecordTypeSymbol) typeSymbol;
+                List<RecordFieldSymbol> recordFields = CommonUtil.getMandatoryRecordFields(includedRecordType);
+                recordFields.forEach(recordFieldSymbol -> {
+                    Optional<String> fieldName = recordFieldSymbol.getName();
+                    TypeSymbol fieldType = recordFieldSymbol.typeDescriptor();
+                    String defaultValue = CommonUtil.getDefaultValueForType(fieldType).orElse("");
+                    if (fieldName.isEmpty() || fieldName.get().isEmpty() ||
+                            existingNamedArgs.contains(fieldName.get())) {
+                        return;
+                    }
```

Comment:
This can also be moved to 136 to avoid calculating `fieldType` and `defaultValue`, if it fails.

Reply:
+1